cmake/backports/FindPythonInterp.cmake

```diff
@@ -54,7 +54,7 @@ endfunction()
 # and choose it if it does. This gives this executable the highest
 # priority, which is expected behaviour.
 find_program(PYTHON_EXECUTABLE python)
```

Comment:
Does this look for an executable called `python`? That won't work on most systems, as `python` is usually Python 2.

I still get Python 3.4 on my system, even though I have 3.6 and 3.7 as well.

Reply:
> Does this look for an executable called python?

> That won't work on most systems, as python is usually Python 2.

It still works, this is only one piece in the "find python" machinery.

> I still get Python 3.4 on my system, even though I have 3.6 and 3.7 as well. 

What is the output of "which python", and "which python3" on your system?

Reply:
```
ulf huvuddator ~ $ which python python3
/usr/bin/python
/usr/bin/python3
ulf huvuddator ~ $ python --version
Python 2.7.15+
ulf huvuddator ~ $ python3 --version
Python 3.6.7
```

Reply:
And here's what CMake says:

```
-- Found PythonInterp: /usr/bin/python3.4 (found suitable version "3.4.9", minimum required is "3.4")`
```

That's what gets used to.

Reply:
hmm, I guess we should have python3 take priority over python3.4.

But I'm a bit busy at the moment, so I don't have time to implement this.

Reply:
Bit borked without it at least, but no panic for me.

Reply:
The patch works fine here.
BTW, if you use virtualenv, `python` will be the version you specify.  So you can't assume the version of `python`.  I agree that the priority should be python -> python3 -> python3.x.  Not sure for 3.4 vs 3.6 vs 3.7 though.

Reply:
Kconfiglib (and probably other things) run faster the more recent the Python version, fwiw.

Reply:
Yes, but the goal is to use the python version that we think the user wants to use. Not the most recent one.

For instance, if a user uses virtualenv to change 'python' into a 3.4 version. Then we want this activated virtualenv to have priority over 'python3.6' (if it happened to be on path).

Reply:
Is there some way to tell CMake which Python version to use? Seems you'd almost always want the most recent one unless you're testing with an older version or something.

Reply:
Could we just document that and then always use the most recent version found?

Reply:
No.

Users expect that if they activate a python installation through virtualenv then it will be used by Zephyr. The implementation must match the user's expectation, not the optimal performance.

Reply:
What does "activate" mean? I never use virtualenv myself.

Reply:
> [activate] will change your $PATH so its first entry is the virtualenv’s bin/ directory. (You have to use source because it changes your shell environment in-place.) This is all it does; it’s purely a convenience.

https://virtualenv.pypa.io/en/latest/userguide/?highlight=activate#activate-script
https://virtualenv.pypa.io/en/latest/

Reply:
Agree to document it but disagree to use the latest.   We can always specify what to use with `-DPYTHON_EXECUTABLE`.  The default should be latest stable interpreter, which is whatever `python3` points to.  Or at least that's what I thought.

CMake version 3.12 introduced `FindPython` which seems to find Python 3.7 even if my python3 points to 3.6. :thinking:  If we are going to use `FindPython` when we require new version of CMake, it might be better to behave the same way?

Reply:
> CMake version 3.12 introduced FindPython which seems to find Python 3.7 even if my python3 points to 3.6. thinking If we are going to use FindPython when we require new version of CMake, it might be better to behave the same way?

Maybe, let's deal with that question when we get there.

Reply:
Feels a bit magic to me to assume that whatever `python3` points to is what's intended to be used. Always use latest is super simple, and it's backwards compatible.

Reply:
This is going to be the priority:

'python'
'python3'
'python3.x'

This makes sense to me. Is simple to understand, and meets the requirements described in: https://github.com/zephyrproject-rtos/zephyr/issues/11857